### PR TITLE
GDB-4837 Set a prefix inline from the mapper table

### PR DESCRIPTION
**Pull request by teodossidossev**

### CODE REVIEW

File: src/app/main/mapper/cell/empty-block/empty-block.component.ts [hunk 1]
```diff
@@ -162,20 +170,37 @@ export class EmptyBlockComponent extends OnDestroyMixin implements OnInit, After
 
   public saveInputValue(emitTab: boolean) {
     let value = this.autoInput.value;
+    let prefixTransformation: string;
+
+    if (this.isExtendedPrefix(value)) {
+      const match = XRegExp.exec(value, this.regex);
+      prefixTransformation = match.namespace;
+      if (match.value) {
+        match.extended ? prefixTransformation += ':' + match.extended + match.extendedEndSymbol : prefixTransformation;
```
Comment (svilenvelikov) [inline src/app/main/mapper/cell/empty-block/empty-block.component.ts:179]: You don't need this ternary operator but just a simple if expression 
```javascript
if (match.extended) { 
  prefixTransformation += ':' + match.extended + match.extendedEndSymbol;
```

File: src/app/main/mapper/iteration/iteration.component.ts [hunk 2]
```diff
@@ -553,6 +553,7 @@ export class IterationComponent extends OnDestroyMixin implements OnInit, AfterV
   public onValueSet(valueSet, triple: any, selected: string, index: number) {
     const value = valueSet.value;
     const source = valueSet.source;
+    let prefixTransformation = valueSet.prefixTransformation;
```
Comment (svilenvelikov) [inline src/app/main/mapper/iteration/iteration.component.ts:556]: Move this on line 590 above the expression where it's first used

File: src/app/main/mapper/iteration/iteration.component.ts [hunk 3]
```diff
@@ -586,19 +587,21 @@ export class IterationComponent extends OnDestroyMixin implements OnInit, AfterV
...
-    const prefix = prefixTransformation && prefixTransformation.prefix;
 
     const data = {
       constant: source === SourceEnum.Constant ? value : undefined,
       columnName: source === SourceEnum.Column ? value : undefined,
       source,
       type: this.getType(selected, triple, value),
       typeMapping: triple.isTypeProperty,
-      expression: prefix ? prefixTransformation.prefix : undefined,
+      expression: prefix ? prefix : undefined,
```
Comment (svilenvelikov) [inline src/app/main/mapper/iteration/iteration.component.ts:604]: Maybe just 
```
expression: prefix,
```

File: src/app/utils/constants.ts [hunk 4]
```diff
@@ -52,5 +52,5 @@ export const SPARQL_AUTOCOMPLETE=`SELECT distinct ?iri ?g WHERE {
 }`;
 export const SOURCE_SIGN = {
   Column: '@',
-  RecordRowID: '#',
+  RecordRowID: '$',
```
Comment (desislava-hristova-ontotext) [inline src/app/utils/constants.ts:55]: Please, update also the en.json

File: cypress/test/edit-mapping.spec.ts [hunk 5]
```diff
@@ -62,6 +62,28 @@ describe('Edit mapping', () => {
...
+      cy.visit('?dataProviderID=ontorefine:123');
+      cy.wait('@loadColumns');
+    });
+
+    // TODO Complete tests when extend functionality is complete
+    it('Should set prefix expression', () => {
+      MappingSteps.getTriples().should('have.length', 1);
+      // And I have created a subject, a predicate and an object
+      MappingSteps.completeTriple(0, 'rdf:extend/subject', 'rdf:extend/@Title', 'rdf:$row_index');
+      MappingSteps.getTripleSubjectSource(0).contains('subject')
```
Comment (desislava-hristova-ontotext) [inline cypress/test/edit-mapping.spec.ts:80]: It's better to check for exact matches, contains me be true, but here the exact values are important.

**Pull request merged**